Docker deployment of tidal-dl-ng

Easel4932 opened the pull request:
I created a simple Dockerfile starting from a Python Image to build a container with tidal-dl-ng installed and settings.json configured with ffmpeg path. Also modified the README.md in consequence to add instructions on how to build the container and how to run it, trying to make it as simple as possible. I hope this addition can be useful to some people and that this PR is done properly. Feel free to educate me on what to modify to do this better if needed :)

exislow commented:
Great idea, thank you very much for this PR. To be honest: In my opinion it is not really efficient to keep a separate `settings.json` in this repository, since it get usually generated by this app on the very first start and you can modify it with `tdn cfg -h`. Do you mind to propose a proper solution in this case? Instead of copying the `settings.json` you could add commands to modify the settings within the `Dockerfile` for example.

Easel4932 commented:
I modified the ```Dockerfile```  to use ```tidal-dl-ng``` cli to configure ffpmeg and download path and removed the ```settings.json```. You were right, this is much cleaner now.

exislow commented on Dockerfile:
```diff
@@ -0,0 +1,35 @@
+RUN apt update 
+
+# Installing nano just in case one need to edit something in the container, installing ffmpeg for tidal-dl-ng
+RUN apt install -y nano ffmpeg
```
Why do you need to install `nano`? This container is hopefully stable and nobody needs to edit anything in it. I would not recommend to install nano to keep the container as small as possible.

Easel4932 replied:
Indeed it is useless, it is a leftover code, from when I was tinkering, that I forgot to remove. I changed the Dockerfile accordingly, thanks for pointing that out! I also wonder if the modifications I made in the README.md are fitting with the rest of it.

exislow reviewed (changes requested):
Great idea. Just some details needs to be tackled.

Easel4932 commented on README.md:
Do you think making Docker its own section is best ? I'm afraid that the README.md would be repetitive with: pip installation then pip usage then docker installation then docker usage. If you have an idea on how to arrange better the README.md, please let me know, I will take those changes into account.

Easel4932 commented:
I made the modifications you suggested (which made the Dockerfile much better honestly), let me know if you want anything else changed.

Vicente015 commented on Dockerfile:
```diff
@@ -0,0 +1,53 @@
+# Creating music folder
+RUN mkdir -p ${MUSIC_PATH}
+
+# Configuring ffmpeg and deownload path fir tidal-dl-ng
```
```suggestion
# Configuring ffmpeg and download path fir tidal-dl-ng
```

found a typo

exislow commented on README.md:
```diff
@@ -42,6 +42,9 @@ If you like this projects and want to support it, feel free to buy me a coffee 
 ## 💻 Installation / Upgrade
 
+
+### 🐍 Installing with pip
```
A blank is missing after `###`.

exislow commented on README.md:
```diff
@@ -50,8 +53,19 @@ pip install --upgrade tidal-dl-ng
+
 ## ⌨️ Usage
 
+### 🐍 Using pip
```
This is misleading. Nobody is using pip here.

exislow commented on README.md:
````diff
@@ -50,8 +53,19 @@ pip install --upgrade tidal-dl-ng
 
+### 🐋 Building the Docker image 
+**Requirements**: Docker == 27.5.0 (other verions might work but are not tested!) 
+All you need is the ```Dockerfile``` file from this repository and be in the same directory as it.
````
Why would you state this, if this file is already there?

exislow commented on README.md:
````diff
@@ -50,8 +53,19 @@ pip install --upgrade tidal-dl-ng
+All you need is the ```Dockerfile``` file from this repository and be in the same directory as it.
+To build the image using ```docker build``` command: 
+```bash
+docker build -t <container-image-name> .
````
Please replace `<container-image-name>` with a suitable name to avoid further errors by the user.

exislow commented on README.md:
````diff
@@ -50,8 +53,19 @@ pip install --upgrade tidal-dl-ng
+```bash
+docker build -t <container-image-name> .
+```
+You can give any name you want to the container. The Dockerfile should be easily modified to fit your needs but works great as is.
````
With a predefined name, you can omit this sentence. Let's keep this readme tight.

little-fish commented on README.md:
````diff
@@ -82,6 +96,19 @@ tidal-dl-ng gui
+
+⚠️ The folder from the host that you map in the container must exist, if Docker creates it it might be owned by ```root``` and tidal-dl-ng **will not have the rights** to write in this folder ⚠️
+
+💡 You can also run a bash shell if you want to tinker in the container. Nano and ffmpeg are installed and the ffmpeg path is preconfigured in the ```settings.json```.
````
Hello. I might have an answer to it.
I guess most of the users, me included, would want to have an image created and stored in their registry. And once they want to download something they just fire up the container and pass http link to it with one single command. They (we) do not want the container to be up and running and consuming any resources if it doesn't do anything. From this point of view it is completely fine to have a settings file (and token file) created on your host (to be easily modifiable outside the image) and just mount them into the container.
But I could be wrong.
